core/src/main/resources/org/apache/spark/ui/static/timeline-view.js

```diff
@@ -50,9 +50,18 @@ function drawApplicationTimeline(groupArray, eventObjArray, startTime, offset) {
        return "#job-" + jobId;
       };
 
+      var getPathForJobEntry = function(baseElem) {
+        var jobIdText = $($(baseElem).find(".application-timeline-content")[0]).text();
+        var jobId = jobIdText.match("\\(Job (\\d+)\\)$")[1];
```

Comment:
`jobIdText` and `jobId` are calculated [here](https://github.com/apache/spark/pull/29757/files#diff-0e05a2e8f77e645ac29df29ee145a8b1R48-R49) so can we reuse them to avoid additional overhead of jQuery selector and regex?

Reply:
Thanks for your comments. i extracted common function `getIdForJobEntry` for selector and regex, and moved it out of `each` loop. It is only triggered by click and hover events. i think this is ok here.